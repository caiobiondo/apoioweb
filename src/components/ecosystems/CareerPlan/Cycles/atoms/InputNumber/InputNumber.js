import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ReactNumberFormat from 'react-number-format';

export default class InputNumber extends Component {
  onValueChange = (value, event) => {
    const { onChange } = this.props;

    if (!onChange) {
      return;
    }

    onChange(value.floatValue || 0, event);
  };

  render() {
    return (
      <ReactNumberFormat
        {...this.props}
        type="tel"
        value={this.props.value || 0}
        onValueChange={this.onValueChange}
        onChange={() => {}}
        isNumericString
        allowNegative={false}
        thousandSeparator="."
        decimalSeparator=","
      />
    );
  }
}

InputNumber.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
  max: PropTypes.number,
  min: PropTypes.number,
  name: PropTypes.string.isRequired,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onKeyPress: PropTypes.func,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  value: PropTypes.any,
};

InputNumber.defaultProps = {
  disabled: false,
  max: 999999,
  placeholder: '',
  required: false,
  type: 'text',
};
