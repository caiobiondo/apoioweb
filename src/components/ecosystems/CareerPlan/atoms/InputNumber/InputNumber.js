import React from 'react';
import PropTypes from 'prop-types';

import InputMask from 'react-input-mask';

const generateMask = (max = 999999) => {
  let generatedMask = '';
  const maxNumberSize = (max + '').length;

  for (let i = maxNumberSize - 1; i >= 0; i--) {
    generatedMask += '9';
  }

  return generatedMask;
};

const BaseInput = props => {
  const mask = generateMask(props.max);

  return (
    <InputMask {...props} value={props.value} onChange={props.onChange} mask={mask} maskChar="" />
  );
};

BaseInput.propTypes = {
  className: PropTypes.string,
  dirty: PropTypes.bool.isRequired,
  disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
  error: PropTypes.array,
  errorMessage: PropTypes.string,
  max: PropTypes.number,
  min: PropTypes.number,
  name: PropTypes.string.isRequired,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onKeyPress: PropTypes.func,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  value: PropTypes.any,
};

BaseInput.defaultProps = {
  dirty: false,
  disabled: false,
  max: 999999,
  placeholder: '',
  required: false,
  type: 'text',
};

export default BaseInput;
