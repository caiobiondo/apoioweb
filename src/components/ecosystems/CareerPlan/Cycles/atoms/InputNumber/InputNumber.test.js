import React from 'react';
import { shallow } from 'enzyme';
import InputNumber from './InputNumber';

const setup = propOverrides => {
  const props = Object.assign(
    {
      name: 'test',
    },
    propOverrides,
  );

  const result = shallow(<InputNumber {...props} />);

  return {
    props,
    result,
  };
};

describe('InputNumber', () => {
  it('should render with the correctly props', () => {
    // given

    // when
    const { result } = setup({ disabled: false });

    // then
    expect(result).toMatchSnapshot();
  });

  describe('onValueChange method', () => {
    it('should call onChange prop with certain values', () => {
      // given
      const onChange = jest.fn();
      const eventMock = { preventDefault: jest.fn() };
      const expectedValue = { floatValue: 1.1 };

      // when
      const { result } = setup({ onChange });
      const instance = result.instance();
      instance.onValueChange(expectedValue, eventMock);

      // then
      expect(onChange).toBeCalledWith(expectedValue.floatValue, eventMock);
    });
  });
});
