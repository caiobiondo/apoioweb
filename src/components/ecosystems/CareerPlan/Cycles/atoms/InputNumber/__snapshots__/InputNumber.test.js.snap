// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber should render with the correctly props 1`] = `
<NumberFormat
  allowNegative={false}
  decimalSeparator=","
  disabled={false}
  displayType="input"
  fixedDecimalScale={false}
  getInputRef={[Function]}
  isAllowed={[Function]}
  isNumericString={true}
  max={999999}
  name="test"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseUp={[Function]}
  onValueChange={[Function]}
  placeholder=""
  prefix=""
  required={false}
  suffix=""
  thousandSeparator="."
  type="tel"
  value={0}
/>
`;
